js/packages/seed-vault/android/src/main/java/com/solanamobile/seedvault/reactnative/SolanaMobileSeedVaultLibModule.kt

```diff
@@ -122,24 +134,87 @@ class SolanaMobileSeedVaultLibModule(val reactContext: ReactApplicationContext)
     }
 
     @ReactMethod
-    fun authorizeNewSeed() {
+    fun requestAuthorizeNewSeed() {
         Log.d(TAG, "Requesting authorization for a new seed...")
         val intent = Wallet.authorizeSeed(WalletContractV1.PURPOSE_SIGN_SOLANA_TRANSACTION)
-        reactContext.currentActivity?.startActivityForResult(intent, REQUEST_AUTHORIZE_SEED_ACCESS);
+        reactContext.currentActivity?.startActivityForResult(intent, REQUEST_AUTHORIZE_SEED_ACCESS)
+    }
+
+    @ReactMethod
+    fun authorizeNewSeed(promise: Promise) {
+        Log.d(TAG, "Requesting authorization for a new seed...")
+        val intent = Wallet.authorizeSeed(WalletContractV1.PURPOSE_SIGN_SOLANA_TRANSACTION)
+        registerForActivityResult(intent, REQUEST_AUTHORIZE_SEED_ACCESS) { resultCode, data -> 
+            try {
+                val authToken = Wallet.onAuthorizeSeedResult(resultCode, data)
+                Log.d(TAG, "Seed authorized, AuthToken=$authToken")
+                
+                promise.resolve(
+                    Arguments.createMap().apply {
+                        putString("authToken", authToken.toString())
+                    }
+                )
+            } catch (e: Wallet.ActionFailedException) {
+                Log.e(TAG, "Seed authorization failed", e)
+                promise.reject(e)
+            }
+        }
     }
 
     @ReactMethod
-    fun createNewSeed() {
+    fun requestCreateNewSeed() {
         Log.d(TAG, "Requesting creation of a new seed...")
         val intent = Wallet.createSeed(WalletContractV1.PURPOSE_SIGN_SOLANA_TRANSACTION)
-        reactContext.currentActivity?.startActivityForResult(intent, REQUEST_CREATE_NEW_SEED);
+        reactContext.currentActivity?.startActivityForResult(intent, REQUEST_CREATE_NEW_SEED)
+    }
+
+    @ReactMethod
+    fun createNewSeed(promise: Promise) {
+        Log.d(TAG, "Requesting creation of a new seed...")
+        val intent = Wallet.createSeed(WalletContractV1.PURPOSE_SIGN_SOLANA_TRANSACTION)
+        registerForActivityResult(intent, REQUEST_CREATE_NEW_SEED) { resultCode, data -> 
+            try {
+                val authToken = Wallet.onCreateSeedResult(resultCode, data)
+                Log.d(TAG, "Seed created, AuthToken=$authToken")
+                
+                promise.resolve(
+                    Arguments.createMap().apply {
+                        putString("authToken", authToken.toString())
+                    }
+                )
+            } catch (e: Wallet.ActionFailedException) {
+                Log.e(TAG, "Seed creation failed", e)
+                promise.reject(e)
+            }
+        }
+    }
+
+    @ReactMethod
+    fun requestImportExistingSeed() {
```

Comment:
Can we remove this now that we have the async/await version in `importExistingSeed`?

Reply:
We could, yes. but Id like to leave it. There is no harm in leaving these in the native module, and they may be useful. we can revisit when ew get to a 1.0 release state.